BlueDBOnDisk/src/main/java/org/bluedb/disk/BlueDbOnDiskBuilder.java

```diff
@@ -22,25 +26,49 @@ public BlueDbOnDiskBuilder withPath(Path path) {
 		return this;
 	}
 
+	/**
+	 * Enables encryption with a given encryption config
+	 *
+	 * @param encryptionConfig the encryption config specifying how to encrypt and decrypt data
+	 * @return itself with the encryption config set
+	 */
+	public BlueDbOnDiskBuilder enableEncryption(EncryptionConfig encryptionConfig) {

```

Comment:
Oh, something that just came to mind. We need to be able to turn encryption on and off anytime we want. My original plan was for `getCurrentEncryptionKeyVersion` returning null to mean that it should not be encrypted. That way if a Chronicall user changes the database encryption setting then Chronicall can just update its encryption key service implementation to start returning something different for `getCurrentEncryptionKeyVersion` and BlueDB would automatically update the next time it starts reading or writing a file. This change kind of has you commit to either encrypting or not encrypting when you first create the database.

Reply:
One option would be to add a `shouldEncryptNewData` method that allows BlueDB to check if it should encrypt or not rather than using a null return value. Maybe something better would be to have the encryption service have a method called `getCurrentEncryptionKeyVersionInfo` or something like that. It could return an object that has information about whether we should encrypt and if so what version to use. These strategies would allow us to easily change the behavior without having an awkward interface. Any thoughts?

Reply:
I still think its fine to assume that if they don't give us an encryption service implementation then we just don't encrypt or decrypt anything.